### PR TITLE
DAOS-14985 test: Address unit test failure due to merge

### PR DESCRIPTION
**Pull request by tanabarr**
Bad merge involving PR-13614 resulted in broken bdevScanEngine go unit test.

- Fix merge conflict affecting unit test by adding roles to bdev in scan.
- Improve reliability of unit test by sorting response content.
- Move filtering of bdevs in bad state so that retry logic is not inadvertently triggered.

Features: control
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by mjmac — DISMISSED** [review]

**Review by kjacque — DISMISSED** [review]

File: src/control/server/instance_storage_rpc.go [hunk 1]
```diff
@@ -230,6 +227,12 @@ func scanEngineBdevsOverDrpc(ctx context.Context, engine Engine, pbReq *ctlpb.Sc
 			Rank:             engineRank.Uint32(),
 		}
 
+		if !sd.Ctrlr.IsScannable() {
```
Comment (knard-intel) [inline src/control/server/instance_storage_rpc.go:230]: Not sure to understand why you have to move this code ?

**Review by knard-intel — APPROVED** [review]
LGTM, just one minor question

**Review by kjacque — APPROVED** [review]

**Comment by tanabarr** [comment]
@knard-intel @mjmac could I get another review on this please?

File: src/control/server/instance_storage_rpc.go (hunk 1 above)
Comment (knard-intel) [reply src/control/server/instance_storage_rpc.go:230]: Thanks for the detailed answer.

**Review by knard-intel — APPROVED** [review]
LGTM

**Comment by tanabarr** [comment]
CI run nr. 4 hit known failure https://daosio.atlassian.net/browse/DAOS-14845

@daos-stack/daos-gatekeeper @gnailzenh Can we please force land this fix

**Pull request merged**